### PR TITLE
COMPAT: drop suppport for python 2.6, #7718

### LINKED ISSUE #7718: COMPAT:  drop support for python 2.6
will close #8760 at same time 

The slowest of the slow of Linux distributions (RHEL) has upgraded their version of Python to 2.7. I think if they're happy with not supporting 2.6, then we can be too. Supporting it is not a *huge* deal, but it would be nice to really encourage people to move forward with upgrading to later versions of Python which would provide the community with momentum towards Python 3. While we can't force everyone to use Python 3, I think dropping support for Python 2.6 is reasonable. The last binary release of Python 2.6 was 2.6.6 and that was in 2010. By now, most people are using python 2.7.

### LINKED ISSUE #8760: TST: require unittest2 for testing on Python 2.6?
There are some nice improvements available in the unittest module on Python 2.7 or later that aren't available on Python 2.6 without using the unittest2 library. In particular, I miss using `assertRaises` as a context manager and `skiptest`.

How about adding the module as a testing requirement on Python 2.6?

### PR DESCRIPTION
**Pull request by jreback**
closes #8760
closes #7718

### CODE REVIEW

**Comment by jreback** [comment]
any complaints, comments, objections?

**Comment by kawochen** [comment]
:smile:  `compat` will get 400 lines shorter

**Comment by jreback** [comment]
oh do I need to things out?

**Comment by kawochen** [comment]
`OrderedDict` and `Counter`

**Comment by TomAugspurger** [comment]
This came up on the scikit-learn mailing list last week. I don't think they've come to a decision yet.

If we are approaching a 1.0 this year we should think about what we want to support indefinitely. I *think* that's an argument for dropping 2.6.

Also, dict comprehension ::raised_hands:

**Comment by jreback** [comment]
yes, I need to make an issue for what we want to *add* (also can use things like ``.assertGreater`` and such in testing).

**Comment by jreback** [comment]
@kawochen turned out to be 478 lines! (unless I took out too much)

File: pandas/compat/__init__.py [hunk 1]
```diff
@@ -268,480 +264,9 @@ def wrapper(cls):
...
-        return result
-
-if sys.version_info[:2] < (2, 7):
-    OrderedDict = _OrderedDict
-    Counter = _Counter
-else:
-    from collections import OrderedDict, Counter
+from collections import OrderedDict, Counter
+from operator import itemgetter
+from heapq import nlargest
```
Comment (kawochen) [inline pandas/compat/__init__.py]: these two lines can be taken out too

Comment (jreback) [inline pandas/compat/__init__.py]: ok

**Comment by jreback** [comment]
puzzled while ``ChainMap`` is not in 2.7. It *says* it has it, but not when I look. oh well.

**Comment by kawochen** [comment]
```
In [115]: from ConfigParser import _Chainmap as ChainMap
```

**Comment by jreback** [comment]
wow, that's an odd place to have it!

**Comment by kawochen** [comment]
If you search for 26, you will find a few more places that can be tweaked (e.g. windows build script)

**Comment by jreback** [comment]
@kawochen thanks.

updated

**Comment by jreback** [comment]
bombs away

**Pull request merged**